src/modules/base.ts

```diff
@@ -1,119 +1,137 @@
-     * @param {string=} cfg.protocolAgentName The name of the agent from which the message will be sent. Default is purecloud-client-app-sdk (name of the package).
-     * @param {string=} cfg.protocolAgentVersion The version of the agent from which the message will be sent. Default is the version of the package.
+     * @param cfg Optional configuration
      *
      * @since 1.0.0
      */
-    constructor(cfg = {}) {
+    constructor(cfg: {
+        /** The origin (protocol, hostname, and port) of the target PC environment (e.g. https://apps.mypurecloud.com). Default is '*'. */
+        targetPcOrigin?: string | null,
+        /** The name of the message protocol under which the message will be sent. Default is purecloud-client-apps. */
+        protocol?: string | null;
+        /** The name of the agent from which the message will be sent. Default is purecloud-client-app-sdk (name of the package). */
+        protocolAgentName?: string | null;
+        /** The version of the agent from which the message will be sent. Default is the version of the package. */
+        protocolAgentVersion?: string | null;
+    } = {}) {
         this._targetPcOrigin = cfg.targetPcOrigin || '*';
 
         this._protocolDetails = {
             name: cfg.protocol || PROTOCOL_NAME,
-            agentName: cfg.protocolAgentName || PROTOCOL_AGENT_NAME,
-            agentVersion: cfg.protocolAgentVersion || PROTOCOL_AGENT_VERSION
+            agentName: cfg.protocolAgentName || __PACKAGE_NAME__,
+            agentVersion: cfg.protocolAgentVersion || __PACKAGE_VERSION__
         };
-
-        /**
-         * Injection point for tests.  Should not be used by api users or extenders.
-         *
-         * @private
-         * @ignore
-         */
-        this._commsUtils = commsUtils;
-
-        // ----- Message Listening
-
-        this._msgListenerCfgs = {};
-        this._msgHandler = this._onMsg.bind(this);
-
-        /**
-         * Injection point for tests.  Should not be used by api users or extenders.
-         *
-         * @private
-         * @ignore
-         */
-        this._myWindow = window;
-
-        /**
-         * Injection point for tests.  Should not be used by api users or extenders.
-         *
-         * @private
-         * @ignore
-         */
-        this._myParent = (window ? window.parent : undefined);
     }
 
     /**
      * Sends the message to purecloud augmenting with environmental details such as
      * target env origin and version info.  Accessible by extenders.
      *
-     * @param {string} actionName
-     * @param {object} msgPayload
-     *
-     * @package
-     * @ignore Extender-use only.  Not a public API
+     * @param actionName
+     * @param msgPayload
      *
      * @since 1.0.0
      */
-    sendMsgToPc(actionName, msgPayload) {
-        this._commsUtils.postMsgToPc(this.buildSdkMsgPayload(...arguments), this._targetPcOrigin);
+    protected sendMsgToPc(actionName: string, msgPayload?: object) {
```

Comment:
I guess according to existing code, this msgPayload type (and the type in buildSdkMsgPayload) be object | null.  But, since it's protected and the existing usages are either undefined or object, seems reasonable not to widen the type unnecessarily.  Was this your thought as well?

Reply:
Yup, definitely with you on null.  Well, either null or undefined on a case-by-case basis, but def not both.

This repo will be a little annoying as we'll always have to be aware of parse boundaries and check types at the door even though the signatures won't technically allow it.  That said, I'm all for keeping the types tight and also using that as our source of truth for what we mostly expect.

Maybe 3.0 can tighten up the types and we can be less lenient than what we currently allow.